### PR TITLE
feat(coupon): update info box for coupon type

### PR DESCRIPTION
**Pull request by StaberindeZA**
## Because

- Once a coupon is successfully applied, we need to show an appropriate
  message depending on coupon type.

## This pull request

- Adds a new hook to determine if a message should be shown and what the
  message should be.

## Issue that this pull request solves

Closes: #11439

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

**Comment by IvoJP** [comment]
Do we need any stories for the case where the duration in months is more than a month or is the test cases you laid out exhaustive enough for our needs? I don't want to get too lost in the weeds/details so I'll leave this to you to judge.

I'm going to hit approve and say LGTM pending your reply. Feel free to update the tests if need be, if you need to change any application code though and want a re-review feel free to request it.

**Review by IvoJP — APPROVED** [review]

**Comment by StaberindeZA** [comment]
> Do we need any stories for the case where the duration in months is more than a month or is the test cases you laid out exhaustive enough for our needs? I don't want to get too lost in the weeds/details so I'll leave this to you to judge.
> 
> I'm going to hit approve and say LGTM pending your reply. Feel free to update the tests if need be, if you need to change any application code though and want a re-review feel free to request it.

I think the existing tests cover the different bases. I did realize though that when the coupon duration and plan interval_count are the same, we should show the same text as for coupon types "once". I've updated the tests and logic for this. (Which also added a test for coupon with duration greater than 1)

I also split out the logic for the one coupon function a bit that hopefully makes that function easier to consume.

**Pull request merged**
